### ISSUE TITLE
Project Status

### ISSUE THREAD

**Issue by xohmz** [rusty-jules/knative-rs#8]
Hi,
I was wondering if you intend to continue growing this project and if you'd like help. I started working on a crate to provide knative types so that I could parse and work with the YAMLs, but ran into your project when I decided to check if someone had already done this. Would you be willing to share where you're at with this project, what issues you've encountered, and what you've learned? I'm not deeply familiar with all the k8s and knative abstractions, but would like to contribute additional type implementations to this project.

Thank,
xohmz

**Comment by rusty-jules** [comment]
Hi @xohmz,

Apologies for the late reply here! Yes this project is still in my sights, it just currently fulfills the requirements of my downstream projects.

The open PRs that have been dangling are mostly there because I don't have a release pipeline in place.

For future work, knative serving seems to be the next logical thing to implement. Instead of handwriting it as was done with eventing, I've tested out [kopium](https://github.com/kube-rs/kopium) and it had great results! However, I don't actually use knative serving, and thus haven't moved forward with that.

In a nutshell, if there's interest then yes there is more to do! I'm open to guiding, sharing, and chatting more.

I learned a lot writing this in-tandem with a k8s controller that serves as an event source, but that is still a private repo since it's very security-oriented (handles OAuth2 refresh tokens) and I have not felt comfortable adding that to the maintenance list.
